### PR TITLE
Agency Filter

### PR DESCRIPTION
**Pull request by mab6bf**

### CODE REVIEW

File: src/js/containers/search/filters/AgencyListContainer.jsx [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+                    title: item.subtier_agency.name,
+                    subtitle: `Sub-Agency of ${item.toptier_agency.name}`,
+                    data: Object.assign({}, item, {
+                        agencyType: 'subtier'
+                    })
+                });
+            });
+
+            // Remove any selected Toptier Agencies
+            const toptierSelections = _.filter(
```
Comment (kevinli-work) [inline src/js/containers/search/filters/AgencyListContainer.jsx]: Can we combine these `_.filter` and `_.remove` into a single `_.remove` using a more complex [predicate](https://lodash.com/docs/4.17.4#remove)?

File: src/js/components/search/filters/agency/AgencyList.jsx [hunk 2]
```diff
@@ -0,0 +1,194 @@
+/**
```
Comment (kevinli-work) [inline src/js/components/search/filters/agency/AgencyList.jsx]: When is this used? I don't see it ever being called.

File: src/js/models/search/queryBuilders/AgencyQuery.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        if (agencyArray.agencyType === 'toptier') {
+            toptierValueSet = _.concat(toptierValueSet, agencyArray.toptier_agency.name);
+        }
+        else {
+            subtierValueSet = _.concat(subtierValueSet, agencyArray.subtier_agency.name);
+        }
+    });
+
+    const toptierFilters = {
+        field: "funding_agency__toptier_agency__name",
```
Comment (mab6bf) [inline src/js/models/search/queryBuilders/AgencyQuery.js]: Fixed to be dynamic

File: src/js/components/search/filters/agency/Agency.jsx [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+                selectedAgencies = this.props.selectedFundingAgencies;
+            }
+            else {
+                selectedAgencies = this.props.selectedAwardingAgencies;
+            }
+
+            return (
+                <div key={`holder-${key}`}>
+                    <AgencyListContainer
+                        key={key}
```
Comment (mab6bf) [inline src/js/components/search/filters/agency/Agency.jsx]: Removed key from AgencyListContainer and changed `div` key to use `type`

File: src/js/components/search/filters/agency/Agency.jsx [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+                        agencyType={type}
+                        selectAgency={this.props.selectAgency}
+                        selectedAgencies={selectedAgencies}
+                        removeAgency={this.props.removeAgency} />
+                    <SelectedAgencies
+                        key={`agencies-${key}`}
```
Comment (mab6bf) [inline src/js/components/search/filters/agency/Agency.jsx]: Removed key from this as well

File: src/js/components/search/filters/agency/AgencyList.jsx (hunk 2 above)
Comment (mab6bf) [inline src/js/components/search/filters/agency/AgencyList.jsx]: Never - Autocomplete replaces this. I'll delete.

File: src/js/components/search/filters/agency/SelectedAgencies.jsx [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+            if (agency.agencyType === 'subtier' &&
+                agency.toptier_agency.name === agency.subtier_agency.name) {
+                label += ' | Sub-Agency';
+            }
+
+            const value = (<ShownAgency
+                agency={agency}
+                label={label}
+                key={key}
+                removeAgency={this.props.removeAgency.bind(
```
Comment (mab6bf) [inline src/js/components/search/filters/agency/SelectedAgencies.jsx]: Done

File: src/js/containers/search/filters/AgencyContainer.jsx [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+export class AgencyContainer extends React.Component {
+    constructor(props) {
+        super(props);
+
+        // Bind functions
+        this.selectAgency = this.selectAgency.bind(this);
+        this.removeAgency = this.removeAgency.bind(this);
+    }
+
+    selectAgency(agency, isValid, agencyType) {
```
Comment (mab6bf) [inline src/js/containers/search/filters/AgencyContainer.jsx]: Good call - combined these into one `toggleAgency` function. I'm keeping the function in `AgencyContainer`, as it's passed down through it's child, `Agency`, which passes it to multiple grandchildren – `AgencyListConatiner` and `SelectedAgencies`. I'd prefer not to duplicate it in both of those, which would also involve converting `SelectedAgencies` to a container.

**Pull request merged**